libpod/events.go

```diff
@@ -77,6 +77,12 @@ func (c *Container) newContainerEventWithInspectData(status events.Status, inspe
 		e.HealthStatus = containerHealthStatus
 	}
 
+	if status == events.Remove {
+		exitCode, _ := c.runtime.state.GetContainerExitCode(c.ID())
+		intExitCode := int(exitCode)
+		e.ContainerExitCode = &intExitCode
```

Comment:
because we now have a proper nil as default you should only set the if there was no error returned by the call.